feature/amazon/src/main/java/com/revenuecat/purchases/amazon/AmazonBilling.kt

```diff
@@ -359,6 +331,54 @@ internal class AmazonBilling constructor(
         }
     }
 
+    private fun queryAmazonPurchases(
+        filterOnlyActivePurchases: Boolean,
+        onSuccess: (Map<String, StoreTransaction>) -> Unit,
+        onError: (PurchasesError) -> Unit
+    ) {
+        executeRequestOnUIThread { connectionError ->
+            if (connectionError == null) {
+                purchaseUpdatesHandler.queryPurchases(
+                    onSuccess = onSuccess@{ receipts, userData ->
+                        val filteredReceipts = if (filterOnlyActivePurchases) {
+                            receipts.filter { it.cancelDate == null || it.cancelDate > dateProvider.now }
+                        } else {
+                            receipts
+                        }
+                        if (filteredReceipts.isEmpty()) {
+                            onSuccess(emptyMap())
+                            return@onSuccess
+                        }
+
+                        getMissingSkusForReceipts(
+                            userData.userId,
+                            filteredReceipts
+                        ) { tokensToSkusMap, errors ->
+                            logErrorsIfAny(errors)
+
+                            if (tokensToSkusMap.isEmpty()) {
+                                val error = PurchasesError(
+                                    PurchasesErrorCode.InvalidReceiptError,
+                                    AmazonStrings.ERROR_FETCHING_PURCHASE_HISTORY_ALL_RECEIPTS_INVALID
+                                )
+                                onError(error)
+                                return@getMissingSkusForReceipts
+                            }
+
+                            val purchasesByHashedToken =
```

Comment:
I don't see an explicit test for it, but if a single sku isn't found for a a token but others are, we return okay and log the error in line 357, correct?

Reply:
There is a test for it, though it doesn't check that the log happens... We currently can't test for logs happening easily. https://github.com/RevenueCat/purchases-android/blob/main/feature/amazon/src/test/java/com/revenuecat/purchases/amazon/AmazonBillingTest.kt#L333 

Lmk if you meant something else though!

Reply:
I wasn't concerned about the log part, I was just making sure that there was a test for it returning okay when a single token isn't found but the rest are. 
Like if a product has been removed from the store so you can't purchase it again, but you did purchase it once and you also purchased other stuff, you should still at least get access to the other stuff while online